sale_manual_delivery/wizard/manual_proc.py

```diff
@@ -63,61 +80,63 @@ def onchange_partner_id(self):
 
     @api.multi
     def record_picking(self):
-        proc_group_obj = self.env["procurement.group"]
+        proc_group_obj = self.env['procurement.group']
+        proc_group_dict = {}
         for wizard in self:
-            carrier_id = wizard.carrier_id if wizard.carrier_id \
-                else wizard.order_id.carrier_id
             date_planned = wizard.date_planned
-            order = wizard.order_id
-            # Use a procurement group according to date planned
-            if not order.procurement_group_id:
-                vals = {
-                    "name": order.name,
-                    "move_type": order.picking_policy,
-                    "sale_id": order.id,
-                    "partner_id": order.partner_shipping_id.id,
-                    "scheduled_date": date_planned,
-                }
-                order.procurement_group_id = proc_group_obj.create(vals)
-                proc_group_to_use = order.procurement_group_id
-            else:
-                proc_group_to_use = self.env['procurement.group'].search(
-                    [
-                        ('sale_id', '=', order.id),
-                        (
-                            'scheduled_date',
-                            '=',
-                            fields.Date.from_string(date_planned)
-                         ),
-                    ], limit=1
-                )
-                if not proc_group_to_use:
-                    proc_group_to_use = order.procurement_group_id.copy({
-                        'scheduled_date': date_planned,
-                    })
-            for line in wizard.line_ids:
-                if line.to_ship_qty > line.ordered_qty - line.existing_qty:
-                    raise UserError(
-                        _(
-                            "You can not deliver more than the "
-                            "remaining quantity. If you need to do "
-                            "so, please edit the sale order first."
-                        )
-                    )
-                if float_compare(line.to_ship_qty, 0, 2):
-                    vals = line.order_line_id._prepare_procurement_values(
-                        group_id=proc_group_to_use
+            for line in wizard.mapped('line_ids.order_line_id'):
+                order = line.order_id
+                if not order.procurement_group_id:
+                    vals = line._prepare_procurement_values()
```

Comment:
I have an error on this when testing:
`TypeError: _prepare_procurement_values() missing 1 required positional argument: 'group_id'`
@aheficent Can you give it a look ?

Reply:
Sure. Could you please tell me what you were doing in Odoo? So I can reproduce and make sure I fix that. Thanks

Reply:
I created a sale order with four lines and tried to create delivery for a partial quantity of two of these lines. I guess it happens as long as the SO has no existing delivery (and therefore no `procurement_group_id`)

Reply:
Thanks for the clarification. I will check asap.

Reply:
I see that group_id field is not required in the original definition of the method _prepare_procurement_values in sale module. However it is in sale_order_dates module (also in Odoo core) :sweat_smile:  I will cover this case.

Reply:
Fixed now.